exp/controllers/alias.go

```diff
@@ -23,13 +23,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 
+	"sigs.k8s.io/cluster-api/controllers/remote"
 	machinepool "sigs.k8s.io/cluster-api/exp/internal/controllers"
 )
 
 // MachinePoolReconciler reconciles a MachinePool object.
 type MachinePoolReconciler struct {
 	Client    client.Client
 	APIReader client.Reader
+	Tracker   *remote.ClusterCacheTracker
```

Comment:
I cannot mark the line here but you also need to set the value in line 43 or else the watch will not work.

Reply:
If the tracker is not set `watchClusterNodes` will return early.

https://github.com/kubernetes-sigs/cluster-api/blob/03b7242ddb7588675306ae5d086a9115ad88c001/exp/internal/controllers/machinepool_controller.go#L307-L309

Reply:
fixed